### ISSUE TITLE
Question : is it possible to implement Message for commonly used crates ?

### ISSUE THREAD

**Issue by Roms1383** [tokio-rs/prost#537]
Hi everybody,
I have a simple question : given how commonly some crates are used when developing, like e.g. [chrono](https://docs.rs/chrono/0.4.19/chrono/), [uuid](https://docs.rs/uuid/0.8.2/uuid/), [nanoid](https://docs.rs/nanoid/0.4.0/nanoid/) (and so on...), would it possible to implement [Message trait](https://docs.rs/prost/0.8.0/prost/trait.Message.html) for them ?
> my apologies if it has been asked before, but I couldn't find mentions of it in the issues search

Currently there's not much alternative when working with them than to create alternate structs and converting back and forth from them ([From/Into](https://doc.rust-lang.org/rust-by-example/conversion/from_into.html) or [CloneFrom/CloneInto](https://crates.io/crates/clone-fields)) or just deal with raw Prost types.

e.g.
```rust
// given a UserProto tonic-build generated type like, e.g.
pub struct UserProto {
    #[prost(string, tag = "1")]
    pub name: ::prost::alloc::string::String,
    #[prost(message, optional, tag = "3")]
    pub created_at: ::core::option::Option<::prost_types::Timestamp>,
    #[prost(message, optional, tag = "4")]
    pub updated_at: ::core::option::Option<::prost_types::Timestamp>,
}
// one often have to create an alternate struct like, e.g.
pub struct User {
    pub name: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: Option<DateTime<Utc>>,
}
// and implement a way to convert from both sides like, e.g.
impl From<UserProto> for User { ... }
impl From<User> for UserProto { ... }
// please also note how cumbersome it is to have to deal with ::core::option::Option<::prost_types::Timestamp>
// for every date when sometimes the date is actually mandatory (e.g. created_at)
```

As a matter of fact, it's indeed way more convenient to locally work with a [chrono::DateTime<Utc>](https://docs.rs/chrono/0.4.19/chrono/struct.DateTime.html) than it is with available [prost_types::Timestamp](https://docs.rs/prost-types/0.8.0/prost_types/struct.Timestamp.html). But when sending over the wire, there's no alternative than to rely on Timestamp.

So long story short, is there any way to get the best of both world, like keeping the prost representation for sending over the wire while deserializing to a more convenient type to work while inside the application itself ?

**Comment by LucioFranco** [comment]
Those crates would need to implement `Message` for their own types and enable it behind a feature flag. Similar to how most crates support `serde`.
